### PR TITLE
Allow a different binary when running docker

### LINKED ISSUE #12135: Allow the use of podman for creating a container-image
**Description**
Currently we have `container-image-jib` and `container-image-docker` for building container images and pushing them to a registry (`container-image-s2i` is openshift specific).

It would be nice to provide a similar experience for `podman` (it can currently be used if the user aliases `docker` to `podman` and used the `container-image-docker` extension).

**Implementation ideas**
The basic question is, do we provide another extension (which would basically do exactly what the docker extension does), or just provide a config switch in `DockerConfig`?

### PR DESCRIPTION
**Pull request by geoand**
This would be the first simple step to supporting podman

Fixes: #12135

### CODE REVIEW

**Review by gsmet — DISMISSED** [review]
I'm approving just to test something, will remove my approval afterwards.

**Review by maxandersen — CHANGES_REQUESTED** [review]

File: extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerConfig.java [hunk 1]
```diff
@@ -40,4 +40,10 @@
      */
     @ConfigItem
     public Optional<List<String>> cacheFrom;
+
+    /**
+     * Name of binary used to execute the docker commands.
+     */
+    @ConfigItem(defaultValue = "docker")
+    public String binaryName;
```
[the remark this answers is not in the document]
Comment (geoand) [reply extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerConfig.java:48]: Good point. I'll go with the later suggestion

**Comment by geoand** [comment]
@maxandersen are you OK with this now?

**Review by maxandersen — APPROVED** [review]
All good.

**Pull request merged**